internal/forwarder/buffnetlink/server.go

```diff
@@ -79,21 +76,106 @@ func (s *Server) ServeMsg(msg *nl.Msg) bool {
 		case gtp5gnl.BUFFER_PACKET:
 			pkt = b[n:int(hdr.Len)]
 		}
+
 		b = b[hdr.Len.Align():]
 	}
 
-	if s.handler != nil && pkt != nil {
-		dldr := report.DLDReport{
-			PDRID:  pdrid,
-			Action: action,
-			BufPkt: pkt,
+	return seid, pdrid, action, pkt, nil
+}
+
+func (s *Server) ServeMsg(msg *nl.Msg) bool {
+	b := msg.Body[genl.SizeofHeader:]
+	var usars map[uint64][]report.USAReport
+
+	for len(b) > 0 {
```

Comment:
This loop is necessary?
```
for len(b) > 0
```
already loop buffer in 'decodbuffer' and 'DecodeAllUSAReports'